### ISSUE TITLE
armv7 version crashes with panic: unaligned 64-bit atomic operation

### ISSUE THREAD

**Issue by akarl10** [livekit/livekit#2383]
When running livekit on armv7 (32bit) livekit crashes as soon as a client connects to the server

**Server**
- Version: [1.5.2]
- Environment: local dev, debian experimental armv7

**Client**
- SDK: element call livekit branch (uses livekit js 0.8.2 as far as I know)
- Version: [0.8.2]

**To Reproduce**
join element call call

**Expected behavior**
simply working

**Additional context**
```
panic: unaligned 64-bit atomic operation

goroutine 39 [running]:
runtime/internal/atomic.panicUnaligned()
        /opt/hostedtoolcache/go/1.21.5/x64/src/runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Xadd64(0x26f51cc, 0x1)
        /opt/hostedtoolcache/go/1.21.5/x64/src/runtime/internal/atomic/atomic_arm.s:258 +0x14
github.com/ua-parser/uap-go/uaparser.(*Parser).ParseOs(0x26f5180, {0x2442050, 0x46})
        /home/runner/go/pkg/mod/github.com/ua-parser/uap-go@v0.0.0-20230823213814-f77b3e91e9dc/uaparser/parser.go:318 +0x17c
github.com/ua-parser/uap-go/uaparser.(*Parser).Parse.func2()
        /home/runner/go/pkg/mod/github.com/ua-parser/uap-go@v0.0.0-20230823213814-f77b3e91e9dc/uaparser/parser.go:249 +0xa0
created by github.com/ua-parser/uap-go/uaparser.(*Parser).Parse in goroutine 36
        /home/runner/go/pkg/mod/github.com/ua-parser/uap-go@v0.0.0-20230823213814-f77b3e91e9dc/uaparser/parser.go:246 +0x1ac
```

I tried a somewhat older version. that one crashed on a different place (probably fixed already), but since I don't see many issues with arm 32 bit I mentioning straight crashes I think nobody really uses arm32? If that is the case and you don't want to waste time with that kind of issues I have no problem with that, but in that case I would suggest removing the arm32 bit build to make everyone aware that only 64bit archs are expected to work/supported

**Comment by akarl10** [comment]
I have some news: I got it working by reordering some structs. For atomics to work they need to be aligned. Only the beginning of a struct is guaranteed to be on a 64bit boundary, so every utils.TimedVersion must be at the front.

there is also a atomic in ua-parser. I will have a look at that project and maybe notify them about that issue (ua-parser will crash on 32bit arm and probably also i386)

**Comment by davidzhao** [comment]
@akarl10 Thank you for chasing this one down! We'd love to make it work for armv7/32. 

Would you be interested in opening a PR to reorganize the structs around?

**Comment by akarl10** [comment]
just for transparency: #2382 fixed that exact stacktrace, but the pr #2386 fixes the atomic alignment in this repository.

With both applied now it seems to work oob, at least it did not crash anymore in my case.
